### DISCUSSION TITLE
Additional Roles

### DISCUSSION THREAD

**Discussion by moritztim** [TheR1D/shell_gpt#390] [Ideas]
I think this project could benefit from a larger selection of roles. Here's some ideas. Feel free to comment more and I'll add them to the list.

- **Image generation Prompt**: generate a prompt to pipe into an image generation AI, kinda like chatgpt with dall-e
#389
- **Code documentation/refactoring**: generate documentation comments for a given code segment or refactor code according to a global user configurable style guide
- **Commit message**: generate a fitting commit message according to a global user configurable style guide
- **Data Access**: A role that knows about/plugs into a document retrieval agent like [llama-index](https://github.com/run-llama/llama_index) would be great. Have sgpt tell you about your notes etc.
https://github.com/TheR1D/shell_gpt/discussions/390#discussioncomment-7879784
- **OpenAI Assistant**: A role that attaches to a specified [OpenAI assistant](https://platform.openai.com/docs/assistants/overview) thread (or creates a new one). One way to build function calling into sgpt.
https://github.com/TheR1D/shell_gpt/discussions/390#discussioncomment-7879788
- **Emoji**: Return a fitting emoji, could be used to search for an emoji for a very specific occasion

**Comment by majorminors** [comment]
A role that knows about/plugs into a document retrieval agent like [llama-index](https://github.com/run-llama/llama_index) would be great. Have sgpt tell you about your notes etc.

**Comment by majorminors** [comment]
A role that attaches to a specified OpenAI assistant thread (or creates a new one). One way to build function calling into sgpt.

**Comment by TheR1D** [comment]
We can create any roles by simple describing them for ShellGPT using `--create-role`.
```shell
sgpt --create-role "git_commit"
# Enter role description: You are git commit message generator, check diff and provide a commit message.
git diff | sgpt --role git_commit
```
Anything else that requires external API calls, or more complex OS operations might be implemented using new function calling feature released in 1.1.0
